WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionDeeplinkActivity.java

```diff
@@ -91,48 +91,41 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
-    protected void onStop() {
-        mDispatcher.unregister(this);
-        super.onStop();
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == RequestCodes.JETPACK_LOGIN && resultCode == RESULT_OK) {
```

Comment:
Nitpick: I'd have preferred the following just to make sure that we don't handle any other request code. There is probably none we could handle right now, but I like making sure the code won't do anything I didn't expect from it:

```
if (requestCode == RequestCodes.JETPACK_LOGIN) {
    if (resultCode == RESULT_OK) {
        trackResult();
    } else {
        finishAndGoBackToSource();
    }
}
```